README.adoc

```diff
@@ -1,3 +1,191 @@
+openstack-cloud: okhttp-3.9.1.jar
+openstack-cloud: openstack4j-okhttp-3.6.jar
+outbound-webhook: okhttp-3.8.1.jar
+pangolin-testrail-connector: okhttp-3.8.1.jar
+performance-signature-dynatrace: okhttp-3.14.4.jar
+performance-signature-dynatracesaas: okhttp-3.14.4.jar
+pipeline-huaweicloud-plugin: okhttp-3.10.0.jar
+protecode-sc: okhttp-3.13.1.jar
+qiniu: okhttp-3.14.4.jar
+qualityclouds: okhttp-3.3.0.jar
+rapid7-insightvm-container-assessment: okhttp-3.6.0.jar
+remoting-kafka: okhttp-3.12.0.jar
+sonar: okhttp-3.7.0.jar
+sweagle: okhttp-3.4.2.jar
+testquality-updater: okhttp-3.5.0.jar
+urbancode-velocity: okhttp-3.5.0.jar
+webhook-eventsource: okhttp-3.3.0.jar
+xframium: okhttp-3.10.0.jar
+yet-another-docker-plugin: okhttp-3.14.4.jar
+yet-another-docker-plugin: docker-java-transport-okhttp-3.2.0.jar
+```
+
+### okhttp-urlconnection 3.x
+
+*High risk, code changes required*
+
+Support for `okhttp-urlconnection` was dropped in 3.14.x and later.
+It only valid in 3.12x and earlier. 
+See the release notes for 3.14.0: https://square.github.io/okhttp/changelog_3x/#version-3140 for temporary helper class workaround.
+
+```
+azure-acs: okhttp-urlconnection-3.11.0.jar
+azure-ad: okhttp-3.11.0.jar
+azure-ad: okhttp-urlconnection-3.11.0.jar
+azure-app-service: okhttp-3.11.0.jar
+azure-app-service: okhttp-urlconnection-3.11.0.jar
+azure-artifact-manager: okhttp-3.14.1.jar
+azure-batch-parallel: okhttp-3.3.1.jar
+azure-batch-parallel: okhttp-urlconnection-3.3.1.jar
+azure-commons: okhttp-3.12.0.jar
+azure-commons: okhttp-urlconnection-3.11.0.jar
+azure-container-agents: okhttp-3.11.0.jar
+azure-container-agents: okhttp-urlconnection-3.11.0.jar
+azure-container-registry-tasks: okhttp-3.12.0.jar
+azure-container-registry-tasks: okhttp-urlconnection-3.11.0.jar
+azure-credentials-ext: okhttp-3.12.6.jar
+azure-credentials: okhttp-3.12.6.jar
+azure-credentials: okhttp-urlconnection-3.12.2.jar
+azure-function: okhttp-3.4.2.jar
+azure-function: okhttp-urlconnection-3.4.2.jar
+azure-iot-edge: okhttp-3.4.2.jar
+azure-iot-edge: okhttp-urlconnection-3.4.2.jar
+azure-vm-agents: okhttp-3.4.2.jar
+azure-vm-agents: okhttp-urlconnection-3.4.2.jar
+azure-vmss: okhttp-3.14.7.jar
+azure-vmss: okhttp-urlconnection-3.4.2.jar
+okhttp-api: okhttp-3.12.12.jar
+okhttp-api: okhttp-urlconnection-3.12.12.jar
+service-fabric: okhttp-3.4.2.jar
+service-fabric: okhttp-urlconnection-3.4.2.jar
+upload-pgyer: okhttp-3.10.0.jar
+upload-pgyer: okhttp-urlconnection-3.10.0.jar
+```
+
+
+### okhttp 2.x
+
+*Low risk, but not recommended*
+
+We can easlily to keep okhttp 2.7.5 (including urlconnection) in the plugin for the foreseeable future. 
+These plugins could be updated to use okhttp-api-plugin with little to no risk of breakages. 
+However, we recommend updating to using okhttp3 before adding this dependency.
+
+A few of these include okhttp-ws.  We will not be including that in the okhttp-api-plugin. We'll leave it to those projects to maintain that version on their own.  
```

Comment:
Sounds good to me, but is there any particular reason for this? Is it only for okhttp 2.x, or does it pull in a bunch of additional dependencies or something?

Reply:
It was purely for package size and surface area, but per our other offline discussion I'm removing okhttp v2.x entirely.